### PR TITLE
BREAKING CHANGE: replace Buffer with Uint8Array

### PR DESCRIPTION
**Pull request by twhy**
# Description

This PR replaces `Buffer` with `Uint8Array` completely in Lumos packages.
`scrypt-js` which requires `Buffer` has been replaced by `@noble/hashes/scrypt`.

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Unit Tests

### CODE REVIEW

File: packages/hd/tests/keystore.test.ts [hunk 1]
```diff
@@ -52,24 +55,56 @@ test("load and check password, loads private key", (t) => {
...
+  // const chainCode = "0x873dff81c02f525623fd1fe5167eac3a55a049de3d314bb42ee227ffed37d508";
+  // const keystore = KeyStore.create(
+  //   new ExtendedPrivateKey(privateKey, chainCode),
+  //   'testpassword',
+  //   {
+  //     iv: bytify("0x8f7097ee22add66106055bf3e241fece"),
+  //     salt: bytify("0x481ead5b0311079d2adcd74f32e3db3bdc5f447bc2217ad8164d2d7ead19079f")
+  //   }
+  // );
   const json = {
```
Comment (homura) [inline packages/hd/tests/keystore.test.ts:105]: Please

- comment out the entire test case without modifying it
- add a new case and paste the permanent URL of the test vector here

Comment (twhy) [reply packages/hd/tests/keystore.test.ts:105]: OK. Done.

File: packages/hd/tests/key.test.ts [hunk 2]
```diff
@@ -51,23 +50,17 @@ test("privateToPublic, derive public key from private key, HexString", (t) => {
...
     );
   });
   t.throws(() => {
-    privateToPublic(
-      Buffer.from(
-        "0xbb39d218506b30ca69b0f3112427877d983dd3cd2cabc742ab723e2964d98016",
-        "hex"
-      )
-    );
+    privateToPublic(bytes.bytify("0x"));
```
Comment (homura) [inline packages/hd/tests/key.test.ts:63]: is there any reason to change the original hex to empty?

Comment (twhy) [reply packages/hd/tests/key.test.ts:63]: In Node.js repl
```ts
> Buffer.from("0xbb39d218506b30ca69b0f3112427877d983dd3cd2cabc742ab723e2964d98016", "hex")
<Buffer >
```

Comment (twhy) [reply packages/hd/tests/key.test.ts:63]: Or we can add one byte to the original hex.
`Old (32 bytes)`: `0xbb39d218506b30ca69b0f3112427877d983dd3cd2cabc742ab723e2964d98016`
`New (33 bytes)`: `0xbb39d218506b30ca69b0f3112427877d983dd3cd2cabc742ab723e2964d9801678`

**Review by homura — APPROVED** [review]

**Pull request merged**